demux/utils/samplesheet.py

```diff
@@ -206,6 +208,65 @@ def _validate_uniq_index(samplesheet):
         return True
 
 
+class MiSeqSamplesheet(Samplesheet):
+
+    def __init__(self, samplesheet_path):
+        self.samplesheet_path = samplesheet_path
+        converted_sheet_path = self.convert_sheet_type(samplesheet_path)
+        self.parse(converted_sheet_path)
+
+    def convert_sheet_type(self, samplesheet_path):
+        """
+        Conver miseq to hiseq style samplesheet for demultiplexing.
+        """
+        dirname = os.path.dirname(samplesheet_path)
```

Comment:
I prefer to use `Path` (pypath) module for all path related operations. Feel free to check it out